addons/l10n_tr_nilvera_einvoice/models/account_edi_xml_ubl_tr.py

```diff
@@ -102,6 +102,20 @@ def _get_partner_party_identification_vals_list(self, partner):
             },
             'id': partner.vat,
         })
+
+        mandatory_categories = self.env["res.partner.category"]._get_l10n_tr_official_mandatory_categories()
+        tr_companies = self.env["res.company"].search([("country_code", "=", "TR"), ("l10n_tr_nilvera_api_key", "!=", False)])
+        if partner in tr_companies.partner_id and not (mandatory_categories & partner.category_id.parent_id):
+            raise UserError(_("Please ensure that your company contact has either the 'MERSISNO' or 'TICARETSICILNO' tag with a value assigned."))
+
+        for category in partner.category_id.filtered(lambda c: c.parent_id in partner.category_id._get_l10n_tr_official_categories()):
```

Comment:
Btw we could do an early continue instead of a filtered since the filtered will loop though the partner.category and then we will loop back on the new record set which is not that great in term of complexity